server/src/domain/system-config/system-config.service.ts

```diff
@@ -53,6 +61,24 @@ export class SystemConfigService {
     this.core.addValidator(validator);
   }
 
+  async handleImmichLatestVersionAvailable() {
+    try {
+      this.logger.debug('Checking if a new version is available ...');
+      const data = await this.repository.getLatestAvailableVersion();
+      this.dateCheckAvailbleVersion = Date.now();
+      if (compareVersions(data.tag_name, serverVersion)) {
+        this.logger.log('New Immich version detected : ' + stringToVersion(data.tag_name).toString());
+        this.availableVersion = stringToVersion(data.tag_name);
+        return true;
+      } else {
+        this.logger.debug('No new version detected');
+      }
+    } catch (error) {
+      this.logger.error('Error occurred:', error);
+      return false;
+    }
+  }
+
```

Comment:
You should return false after the try catch instead of in l. 78. Currently, the method will return undefined if no new version is detected.